File: src/models/ResponseFromCall.ts
export type ResponseFromCall = {
    status: string,
    body: string,
    headers: Map,
    cookies: Map,
    duration: TimeMeasures
}

export interface Map {
    [key: string]: string;
}

export type TimeMeasures = {
    response_duration: String,
    duration: String
}

File: src-tauri/src/models/response_from_call.rs
use std::collections::HashMap;


use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ResponseFromCall {
    pub status: String,
    pub body: String,
    pub headers: HashMap<String, String>,
    pub cookies: HashMap<String, String>,
    pub duration: TimeMeasures,
}

#[derive(Serialize, Deserialize)]
pub struct TimeMeasures{
    pub response_duration: String,
    pub duration: String,
}